File: C programming/eissasharaf-cs50-problems-2020-x-cash/cash.c
//include libraries
#include<stdio.h>
#include<cs50.h>
#include<math.h>

//main function begin
int main(void)
{
    //variable declearation
    int coin = 0, cent; 
    float dollar;
    
    //prompt the user for a non negative change owed
    do
    {
        dollar = get_float("change owed:");
    }
    while (dollar < 0);

//calculating the number of coins
    cent = round(dollar * 100);
    //quarters
    while (cent >= 25)
    {
        cent -= 25; 
        coin++;
    }
    //dimes
    while (cent >= 10)
    {
        cent -= 10; 
        coin++;
    }
    // nickels
    while (cent >= 5)
    {
        cent -= 5; 
        coin++;
    }
    //pennies  
    while (cent >= 1)
    { 
        cent -= 1; 
        coin++;
    }

    //output  
    printf("%i\n", coin);
}  
//end of main

File: python/eissasharaf-cs50-problems-2020-x-sentimental-cash/cash.py
from cs50 import get_float

owed = -1

while owed < 0:
    owed = get_float("Change owed: ")
coin = [25, 10, 5, 1]
total = 0
owed = round(owed * 100)
for i in coin:
    total += int(owed / i)
    owed %= i 
print(total)